operator/api/v1alpha1/condition_builder.go

```diff
@@ -31,24 +33,47 @@ func (cb *ConditionBuilder) SetReason(reason KymaConditionReason) *ConditionBuil
 	return cb
 }
 
+func (cb *ConditionBuilder) SetObservedGeneration(observedGeneration int64) *ConditionBuilder {
+	cb.ObservedGeneration = observedGeneration
+	return cb
+}
+
 func (cb *ConditionBuilder) Build() metav1.Condition {
 	return metav1.Condition{
 		Type:               string(ConditionTypeReady),
 		Status:             cb.Status,
 		LastTransitionTime: metav1.Time{Time: time.Now()},
 		Reason:             string(cb.Reason),
 		Message:            cb.generateMessage(),
+		ObservedGeneration: cb.ObservedGeneration,
 	}
 }
 
 func (cb *ConditionBuilder) generateMessage() string {
-	if cb.Reason == ConditionReasonModulesAreReady {
-		if cb.Status == metav1.ConditionTrue {
+	switch cb.Reason {
+	case ConditionReasonModulesAreReady:
+		switch cb.Status {
+		case metav1.ConditionTrue:
 			return MessageModuleInReadyState
-		}
-		if cb.Status == metav1.ConditionFalse || cb.Status == metav1.ConditionUnknown {
+		case metav1.ConditionUnknown:
+			fallthrough
+		case metav1.ConditionFalse:
+			fallthrough
+		default:
 			return MessageModuleNotInReadyState
 		}
+	case ConditionReasonModuleCatalogIsReady:
+		switch cb.Status {
+		case metav1.ConditionTrue:
+			return MessageModuleCatalogIsSynced
+		case metav1.ConditionUnknown:
+			fallthrough
+		case metav1.ConditionFalse:
+			fallthrough
+		default:
+			return MessageModuleCatalogIsOutOfSync
+		}
+	default:
+		return "no detailed message available as reason is unknown to API"
```

Comment:
should we put this return state also as an const?
If you not use this default case, and return it in the end, then you can get rid of each `fallthrough` in above `case`. That's not a typical usage of fallthrough I think.

Reply:
I tried this and then the linter said no to me :(

Reply:
Should I adjust it so it doesnt use this linter setting anymore?